tmt/cli.py

```diff
@@ -2229,3 +2236,38 @@ def completion_zsh(context: Context, install: bool, **kwargs: Any) -> None:
 def completion_fish(context: Context, install: bool, **kwargs: Any) -> None:
     """ Setup shell completions for fish """
     setup_completion('fish', install, context)
+
+
+@main.command(name='link')
+@pass_context
+@click.argument('link', nargs=1, metavar='[RELATION:]TARGET')
+@click.argument('names', nargs=-1, metavar='[TEST|PLAN|STORY]...')
+@option(
+    '--separate', is_flag=True,
+    help="Create linking separately for multiple passed objects.")
+def link(context: Context,
+         names: list[str],
+         link: str,
+         separate: bool,
+         ) -> None:
+    """
+    Create a link to tmt web service in an issue tracking software.
+
+    Using the specified target, a link will be generated and added
+    to an issue. Link is generated from names of tmt objects
+    passed in arguments and configuration file.
+    """
+
+    tmt_objects = (
+        context.obj.tree.tests(names=list(names)) +
+        context.obj.tree.plans(names=list(names)) +
+        context.obj.tree.stories(names=list(names)))
```

Comment:
```suggestion
        context.obj.tree.tests(names=names.copy()) +
        context.obj.tree.plans(names=names.copy()) +
        context.obj.tree.stories(names=names.copy()))
```

Reply:
Not necessarily, Click delivers sequences as tuples, `list()` is needed for safe conversion. The type should reflect that, I suppose, `names: Union[list[str], tuple[str, ...]]`?

Reply:
Ack, thanks.

Reply:
This would not work as `names` are `tuple`:

    'tuple' object has no attribute 'copy'

Conversion to `list` is needed to actually concatenate tests, plans and stories:

    can only concatenate tuple (not "list") to tuple

So, I guess, we're good with the current approach?

Reply:
Yes yes, I was wrongly assuming they are lists, apologies.